src/vis/vis-view.js

```diff
@@ -20,6 +20,7 @@ var Vis = View.extend({
     this.overlays = [];
 
     this.model.on('change:showLegends', this._onShowLegendsChanged, this);
+    this.model.on('change:showLayerSelector', this._onShowLayerSelectorChanged, this);
```

Comment:
Do we need this? From what I understand, `legendsView` is going to be rendered no mater what... I see `legendsView` already hides itself when these two flags are false...